internal/runners/show/show.go

```diff
@@ -46,86 +63,102 @@ func New(prime primeable) *Show {
+			return locale.WrapError(fail.ToError(), "err_show_parse_namespace", "Invalid remote argument, must be of the form <Owner>/<Project>")
 		}
 
-		if _, err := os.Stat(projectFilePath); err != nil {
-			return locale.WrapError(
-				err,
-				"err_state_show_no_config",
-				"activestate.yaml file not found in the given location.",
-			)
+		owner = namespaced.Owner
+		projectName = namespaced.Project
+	} else {
+		if s.project == nil {
+			return locale.NewError("err_no_projectfile")
 		}
 
-		projectFile, err := projectfile.Parse(projectFilePath)
+		owner = s.project.Source().Owner
+		projectName = s.project.Source().Name
+		projectURL = s.project.Source().Project
+
+		events, err = eventsData(s.project.Source(), s.conditional)
 		if err != nil {
-			logging.Errorf("Unable to parse activestate.yaml: %s", err)
-			return locale.WrapError(
-				err,
-				"err_state_show_project_parse",
-				"Could not parse activestate.yaml.",
-			)
+			return locale.WrapError(err, "err_show_events", "Could not parse events")
 		}
 
-		var fail *failures.Failure
-		pj, fail = prj.New(projectFile)
-		if fail != nil {
-			return fail.ToError()
+		scripts, err = scriptsData(s.project.Source(), s.conditional)
+		if err != nil {
+			return locale.WrapError(err, "err_show_scripts", "Could not parse scripts")
 		}
 	}
 
-	src := pj.Source()
+	remoteProject, fail := model.FetchProjectByName(owner, projectName)
+	if fail != nil && fail.Type.Matches(model.FailProjectNotFound) {
+		return locale.WrapError(fail.ToError(), "err_show_project_not_found", "Please run `state push` to synchronize this project with the ActiveState Platform.")
+	} else if fail != nil {
+		return locale.WrapError(err, "err_show_get_project", "Could not get remote project details")
+	}
 
-	updater.PrintUpdateMessage(src.Path())
+	branch, fail := model.DefaultBranchForProjectName(owner, projectName)
+	if fail != nil {
+		return locale.WrapError(fail.ToError(), "err_show_get_default_branch", "Could not get project information from the platform")
+	}
+	if branch.CommitID == nil {
+		return locale.NewError("err_show_commitID", "Remote project details are incorrect. Default branch is missing commitID")
+	}
 
-	events, err := eventsData(src, s.conditional)
+	if projectURL == "" {
+		projectURL = model.ProjectURL(owner, projectName, branch.CommitID.String())
+	}
+
+	platforms, err := platformsData(owner, projectName, *branch.CommitID)
 	if err != nil {
-		return locale.WrapError(err, "err_show_events", "Could not parse events.")
+		return locale.WrapError(err, "err_show_platforms", "Could not retrieve platform information")
 	}
 
-	scripts, err := scriptsData(src, s.conditional)
+	languages, err := languagesData(owner, projectName)
 	if err != nil {
-		return locale.WrapError(err, "err_show_scripts", "Could not parse scripts.")
+		return locale.WrapError(err, "err_show_langauges", "Could not retrieve language information")
+	}
+
+	commit, err := commitsData(owner, projectName, *branch.CommitID, s.project)
```

Comment:
This should be prioritizing the commit set in the projectfile, otherwise the "x behind" will always be zero.

Reply:
Updated the function for this